examples/transformerconfigs/README.md

```diff
@@ -119,44 +119,472 @@ commonAnnotations:
 
 ## Name reference transformer
 
-Name reference transformer's configuration is different from all other transformers. It contains a list of `nameReferences`, which represent all of the possible fields that a type could be used as a reference in other types of resources. A `nameReference` contains a type such as ConfigMap as well as a list of `fieldSpecs` where ConfigMap is referenced in other resources. Here is an example:
+`nameReference` Transformer is used to tie a target resource's name to a list of other resources' referrers' names.
+Once tied, the referrers' names will change alongside the target name via transformers like `namePrefix` and `nameSuffix`
 
+### Usage
+  - The syntax `nameReference` should be written in the `configurations` files, not directly in `kustomization.yaml`
+  - kustomize has a set of builtin nameReference, and you don't need to write additional configs to use those nameReference. Check the full list [here](#buildin-namereference).
+  - The referrer's `name` has to match the target's `name`. Otherwise, the referrer's name won't be changed.
```

Comment:
```suggestion
  - The referrer's `name` has to match the target's `name`. Otherwise, the referrer's name won't be changed. This name can be the target's current name, or, if the target has been through multiple name transformations, can be any of the target's previous names.
```